### PR TITLE
Refactor the multiplication operators in CartesianState and derived classes

### PR DESCRIPTION
**Pull request by buschbapti**
## Description

- #22 

This PR refactors the multiplication operators in C++ to address the rules agreed in #22.

It also includes the following:
- refactor the docstrings to follow the guidelines introduced in #23
- move the vector transformation from CartesianPose to CartesianWrench (to be consistent with the rules in #22)
- adds the missing scaling operation in CartesianWrench

## Review guidelines

Estimated Time of Review: 15 minutes

### Blocked by:
- #20 

As long as #20 is not resolved, the tests are not passing.

## Additional comments

The scaling operations with a 6x6 matrix are not well defined. At the moment, we treat linear and angular operations independently. Therefore, this should be a scaling operation with a `std:pair<Eigen::Matrix3d, Eigen::Matrix3d>` but that is a bit verbose. Also, the same operations are not defined for `CartesianPose`. Another PR should probably address those separately, thus commit 6f82e4e711a331bc7b891d8c4d8ec138312e9320 could be reversed and included there.

### CODE REVIEW

**Comment by buschbapti** [comment]
It seems the rebase on the operator branch broke the commit history. PR #23 that was merged now appear here as well. I will try to rebase and solve that.

File: source/state_representation/test/tests/space/cartesian/test_cartesian_state.cpp [hunk 1]
```diff
@@ -520,6 +520,93 @@ TEST(CartesianStateTest, Truthiness) {
...
+TEST(CartesianStateTest, TestMultiplicationOperators) {
+  CartesianState state = CartesianState::Random("world");
+  CartesianPose pose = CartesianPose::Random("world");
+  CartesianTwist twist = CartesianTwist::Random("world");
+  CartesianAcceleration acc = CartesianAcceleration::Random("world");
+  CartesianWrench wrench = CartesianWrench::Random("world");
+
+  // CartesianState multiplied with any derived stays a CartesianState
+  auto r1 = state * state;
+  EXPECT_TRUE(r1.get_type() == StateType::CARTESIAN_STATE);
```
Comment (eeberhard) [inline source/state_representation/test/tests/space/cartesian/test_cartesian_state.cpp:532]: nitpick: any test case that does `EXPECT_TRUE(a == b)` can be reduced to `EXPECT_EQ(a, b)`. The benefit of the latter is more informative output on test failure.
```suggestion
  EXPECT_EQ(r1.get_type(), StateType::CARTESIAN_STATE);
```

**Review by eeberhard — APPROVED** [review]
It took a bit of time to go through but the test summarizes the expected behavior very well. The docstrings are really much better now.

With proper type setting after value-casting, the tests should also pass, so it seems good to go in to the refactor branch now.

**Comment by buschbapti** [comment]
Thanks @eeberhard. Comments have been addressed and the version updated. I will close this one before the subtraction.

File: source/state_representation/test/tests/space/cartesian/test_cartesian_state.cpp [hunk 2]
```diff
@@ -520,6 +520,93 @@ TEST(CartesianStateTest, Truthiness) {
   EXPECT_TRUE(empty);
 }
 
+TEST(CartesianStateTest, TestMultiplicationOperators) {
+  CartesianState state = CartesianState::Random("test");
```
Comment (domire8) [inline source/state_representation/test/tests/space/cartesian/test_cartesian_state.cpp:524]: This is not gonna work for the reason that CartesianState(test, world)  is not compatible with CartesianPose(test, world), which is why i had them as CartesianState(world, world) initially

Comment (eeberhard) [reply source/state_representation/test/tests/space/cartesian/test_cartesian_state.cpp:524]: Good point, of course it was different for the addition operators. It's difficult to catch that behavior when you already expect the tests to fail ;)

Comment (buschbapti) [reply source/state_representation/test/tests/space/cartesian/test_cartesian_state.cpp:524]: That is indeed a good point thanks for checking that. I am doing division operator next and will correct it then.

**Pull request merged**